Further reduce use of WTF_ALLOW_UNSAFE_BUFFER_USAGE in WTF/

cdumez opened the pull request:
#### 83a834b1ac4f39bd80bb986051d2ad80af7d23ea
<pre>
Further reduce use of WTF_ALLOW_UNSAFE_BUFFER_USAGE in WTF/
<a href="https://bugs.webkit.org/show_bug.cgi?id=283252">https://bugs.webkit.org/show_bug.cgi?id=283252</a>

Reviewed by Darin Adler.

* Source/WTF/wtf/SHA1.cpp:
(WTF::SHA1::addUTF8Bytes):
* Source/WTF/wtf/SixCharacterHash.cpp:
(WTF::integerToSixCharacterHashString):
* Source/WTF/wtf/StackBounds.cpp:
(WTF::StackBounds::newThreadStackBounds):
(WTF::StackBounds::currentThreadStackBoundsInternal):
* Source/WTF/wtf/StringPrintStream.cpp:
(WTF::StringPrintStream::StringPrintStream):
(WTF::StringPrintStream::~StringPrintStream):
(WTF::StringPrintStream::vprintf):
(WTF::StringPrintStream::toCString const):
(WTF::StringPrintStream::reset):
(WTF::StringPrintStream::tryToString const):
(WTF::StringPrintStream::toString const):
(WTF::StringPrintStream::toStringWithLatin1Fallback const):
(WTF::StringPrintStream::increaseSize):
(WTF::StringPrintStream::toCString): Deleted.
(WTF::StringPrintStream::tryToString): Deleted.
(WTF::StringPrintStream::toString): Deleted.
(WTF::StringPrintStream::toStringWithLatin1Fallback): Deleted.
* Source/WTF/wtf/StringPrintStream.h:
* Source/WTF/wtf/dtoa/double-conversion.cc:
* Source/WTF/wtf/text/EscapedFormsForJSON.h:
* Source/WTF/wtf/text/StringBuilderJSON.cpp:
(WTF::StringBuilder::appendQuotedJSONString):
* Source/WTF/wtf/text/StringBuilderJSON.h:
(WTF::appendEscapedJSONStringContent):
* Source/WTF/wtf/text/StringSearch.h:
(WTF::BoyerMooreHorspoolTable::initializeTable):
(WTF::BoyerMooreHorspoolTable::findInner const):

Canonical link: <a href="https://commits.webkit.org/286712@main">https://commits.webkit.org/286712@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b9057a5d2117df3e7805acee3edb25441e62796f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/76880 "3 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/55915 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/29786 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/81422 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/28156 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/65057 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/4208 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/60240 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/18322 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/79947 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/50190 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/65997 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/40557 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/47592 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/23492 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/26481 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/70063 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/68709 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/23821 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/82860 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/76157 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/4256 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/2830 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/68518 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/4410 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/65970 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/67773 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/11758 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/9841 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/98412 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11899 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/4203 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/21532 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/4226 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/7657 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/5984 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WTF/wtf/StringPrintStream.cpp:
```diff
@@ -28,113 +28,107 @@
-    , m_next(0)
-    , m_size(sizeof(m_inlineBuffer))
 {
+    m_buffer = std::span<char> { m_inlineBuffer };
```
Is the `<char>` needed?

cdumez replied:
I'll try but usually yes. Because otherwise you end up with a `std::span<char, ARRAY_SIZE>` when constructing from an array.

darinadler commented on Source/WTF/wtf/text/StringBuilderJSON.h:
```diff
@@ -15,60 +15,72 @@
+                output[0] = upperNibbleToLowercaseASCIIHexDigit(character);
+                output = output.subspan(1);
+                output[0] = lowerNibbleToLowercaseASCIIHexDigit(character);
+                output = output.subspan(1);
```
```suggestion
                output[0] = '0';
                output[1] = '0';
                output[2] = upperNibbleToLowercaseASCIIHexDigit(character);
                output[3] = lowerNibbleToLowercaseASCIIHexDigit(character);
                output = output.subspan(4);
```

darinadler commented on Source/WTF/wtf/text/StringBuilderJSON.h:
```diff
@@ -15,60 +15,72 @@
+            output[0] = '\\';
+            output = output.subspan(1);
+            output[0] = escaped;
+            output = output.subspan(1);
```
```suggestion
            output[0] = '\\';
            output[1] = escaped;
            output = output.subspan(2);
```

darinadler commented on Source/WTF/wtf/text/StringBuilderJSON.h:
```diff
@@ -15,60 +15,72 @@
+                output[0] = character;
+                output = output.subspan(1);
+                output[0] = next;
+                output = output.subspan(1);
```
```suggestion
                output[0] = character;
                output[1] = next;
                output = output.subspan(2);
```

darinadler commented on Source/WTF/wtf/text/StringBuilderJSON.h:
```diff
@@ -15,60 +15,72 @@
+        output[0] = upperNibbleToLowercaseASCIIHexDigit(lower);
+        output = output.subspan(1);
+        output[0] = lowerNibbleToLowercaseASCIIHexDigit(lower);
+        output = output.subspan(1);
```
```suggestion
        output[0] = '\\';
        output[1] = 'u';
        output[2] = upperNibbleToLowercaseASCIIHexDigit(upper);
        output[3] = lowerNibbleToLowercaseASCIIHexDigit(upper);
        output[4] = upperNibbleToLowercaseASCIIHexDigit(lower);
        output[5] = lowerNibbleToLowercaseASCIIHexDigit(lower);
        output = output.subspan(6);
```

cdumez committed:
Further reduce use of WTF_ALLOW_UNSAFE_BUFFER_USAGE in WTF/
https://bugs.webkit.org/show_bug.cgi?id=283252

Reviewed by Darin Adler.

* Source/WTF/wtf/SHA1.cpp:
(WTF::SHA1::addUTF8Bytes):
* Source/WTF/wtf/SixCharacterHash.cpp:
(WTF::integerToSixCharacterHashString):
* Source/WTF/wtf/StackBounds.cpp:
(WTF::StackBounds::newThreadStackBounds):
(WTF::StackBounds::currentThreadStackBoundsInternal):
* Source/WTF/wtf/StringPrintStream.cpp:
(WTF::StringPrintStream::StringPrintStream):
(WTF::StringPrintStream::~StringPrintStream):
(WTF::StringPrintStream::vprintf):
(WTF::StringPrintStream::toCString const):
(WTF::StringPrintStream::reset):
(WTF::StringPrintStream::tryToString const):
(WTF::StringPrintStream::toString const):
(WTF::StringPrintStream::toStringWithLatin1Fallback const):
(WTF::StringPrintStream::increaseSize):
(WTF::StringPrintStream::toCString): Deleted.
(WTF::StringPrintStream::tryToString): Deleted.
(WTF::StringPrintStream::toString): Deleted.
(WTF::StringPrintStream::toStringWithLatin1Fallback): Deleted.
* Source/WTF/wtf/StringPrintStream.h:
* Source/WTF/wtf/dtoa/double-conversion.cc:
* Source/WTF/wtf/text/EscapedFormsForJSON.h:
* Source/WTF/wtf/text/StringBuilderJSON.cpp:
(WTF::StringBuilder::appendQuotedJSONString):
* Source/WTF/wtf/text/StringBuilderJSON.h:
(WTF::appendEscapedJSONStringContent):
* Source/WTF/wtf/text/StringSearch.h:
(WTF::BoyerMooreHorspoolTable::initializeTable):
(WTF::BoyerMooreHorspoolTable::findInner const):

Canonical link: https://commits.webkit.org/286712@main
